### PR TITLE
[Elastic Agent] Pick up version from libbeat

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

For historical reasons we were keeping track of our version ourselves which creates overhead for releases. Picking up version from libbeat keeps us in par with other beats we're releasing with

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -221,6 +221,7 @@ github.com/eapache/queue v1.1.0 h1:YOEu7KNc61ntiQlcEeUIoDTJ2o8mQznoNvUhiigpIqc=
 github.com/eapache/queue v1.1.0/go.mod h1:6eCeP0CKFpHLu8blIFXhExK/dRa7WDZfr6jVFPTqq+I=
 github.com/eclipse/paho.mqtt.golang v1.2.1-0.20200121105743-0d940dd29fd2 h1:DW6WrARxK5J+o8uAKCiACi5wy9EK1UzrsCpGBPsKHAA=
 github.com/eclipse/paho.mqtt.golang v1.2.1-0.20200121105743-0d940dd29fd2/go.mod h1:H9keYFcgq3Qr5OUJm/JZI/i6U7joQ8SYLhZwfeOo6Ts=
+github.com/elastic/beats v7.6.2+incompatible h1:jHdLv83KURaqWUC6f55iMyVP6LYZrgElfeqxKWcskVE=
```
Comment (michalpristas) [inline go.sum:224]: oops

**Comment by michalpristas** [comment]
we started out in a separate repo without libbeat so we implemented everything from scratch without dependencied we dont need. even this small version command instantiates a beat Instance,something agent does not use.

**Comment by ph** [comment]
@blakerouse We have instead started out outside of beats and we wanted minimal dependencies on libbeat, after a few iterations maybe this was not the best decision because. We should try to converge on beats on behavior. Its something  we should discuss moving forward.

One thing that might also happens is us to move to a separate repository.

File: x-pack/elastic-agent/magefile.go [hunk 2]
```diff
@@ -340,8 +340,8 @@ func flagsSet() []string {
 	isSnapshot, _ := os.LookupEnv(snapshotEnv)
 
 	return []string{
-		fmt.Sprintf(`-X "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.buildTime=%s"`, ts),
-		fmt.Sprintf(`-X "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.commit=%s"`, commitID),
+		fmt.Sprintf(`-X "github.com/elastic/beats/v7/libbeat/version.buildTime=%s"`, ts),
+		fmt.Sprintf(`-X "github.com/elastic/beats/v7/libbeat/version.commit=%s"`, commitID),
 		fmt.Sprintf(` -X "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.snapshot=%s"`, isSnapshot),
```
Comment (ph) [inline x-pack/elastic-agent/magefile.go:345]: Do we really need this now? I would assume this would be handled by the libbeat part?

Comment (michalpristas) [reply x-pack/elastic-agent/magefile.go:345]: you're right i will try it to make sure

**Comment by michalpristas** [comment]
brought even with libbeat which removed the ability to override version using dev subcommand, i dont think we need that anymore with restrictions from kibana removed

**Comment by ph** [comment]
@michalpristas I think this breaks the packaging?

```
Error: running "go build -o build/golang-crossbuild/elastic-agent-linux-arm64 -ldflags -s "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.snapshot=false" -X github.com/elastic/beats/v7/libbeat/version.buildTime=2020-05-11T15:37:12Z -X github.com/elastic/beats/v7/libbeat/version.commit=3bcfa2a9984ad75e39a169e0f1d929a2b7f17c6c" failed with exit code 2
Error: failed building for linux/arm64: exit status 2
failed building for linux/arm64: exit status 2
# github.com/elastic/beats/v7/x-pack/elastic-agent
usage: link [options] main.o
  -B note
    	add an ELF NT_GNU_BUILD_ID note when using ELF
  -E entry
    	set entry symbol name
  -H type
    	set header type
  -I linker
    	use linker as ELF dynamic linker
  -L directory
    	add specified directory to library path
  -R quantum
    	set address rounding quantum (default -1)
  -T address
    	set text segment address (default -1)
  -V	print version and exit
  -X definition
    	add string value definition of the form importpath.name=value
  -a	disassemble output
  -buildid id
    	record id as Go toolchain build id
  -buildmode mode
    	set build mode
  -c	dump call graph
  -compressdwarf
    	compress DWARF if possible (default true)
  -cpuprofile file
    	write cpu profile to file
  -d	disable dynamic executable
  -debugtramp int
    	debug trampolines
  -dumpdep
    	dump symbol dependency graph
  -extar string
    	archive program for buildmode=c-archive
  -extld linker
    	use linker when linking in external mode
  -extldflags flags
    	pass flags to external linker
  -f	ignore version mismatch
  -g	disable go package data checks
  -h	halt on error
  -importcfg file
    	read import configuration from file
  -installsuffix suffix
    	set package directory suffix
  -k symbol
    	set field tracking symbol
  -libgcc string
    	compiler support lib for internal linking; use "none" to disable
  -linkmode mode
    	set link mode
  -linkshared
    	link against installed Go shared libraries
  -memprofile file
    	write memory profile to file
  -memprofilerate rate
    	set runtime.MemProfileRate to rate
  -msan
    	enable MSan interface
  -n	dump symbol table
  -o file
    	write output to file
  -pluginpath string
    	full path name for plugin
  -r path
    	set the ELF dynamic linker search path to dir1:dir2:...
  -race
    	enable race detector
  -s	disable symbol table
  -strictdups int
    	sanity check duplicate symbol contents during object file reading (1=warn 2=err).
  -tmpdir directory
    	use directory for temporary files
  -u	reject unsafe packages
  -v	print link trace
  -w	disable DWARF generation
Error: running "go build -o build/golang-crossbuild/elastic-agent-darwin-amd64 -ldflags -s "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.snapshot=false" -X github.com/elastic/beats/v7/libbeat/version.buildTime=2020-05-11T15:37:11Z -X github.com/elastic/beats/v7/libbeat/version.commit=3bcfa2a9984ad75e39a169e0f1d929a2b7f17c6c" failed with exit code 2
Error: failed building for darwin/amd64: exit status 2
failed building for darwin/amd64: exit status 2
package ran for 3m15.423462598s
Error: running "docker run --env EXEC_UID=1000 --env EXEC_GID=985 --env GOFLAGS=-mod=vendor --rm --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= --env SNAPSHOT=false -v /home/ph/go/src/github.com/elastic/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/x-pack/elastic-agent docker.elastic.co/beats-dev/golang-crossbuild:1.13.10-main-debian7 --build-cmd build/mage-linux-amd64 golangCrossBuild -p linux/amd64" failed with exit code 1
running "docker run --env EXEC_UID=1000 --env EXEC_GID=985 --env GOFLAGS=-mod=vendor --rm --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= --env SNAPSHOT=false -v /home/ph/go/src/github.com/elastic/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/x-pack/elastic-agent docker.elastic.co/beats-dev/golang-crossbuild:1.13.10-main --build-cmd build/ma

**Comment by michalpristas** [comment]
@ph LDFlags vs Vars sorry, fixed

**Pull request merged**